Add mongo

Javier-Rotelli opened the pull request.

Javier-Rotelli committed:
agrego mongo

Javier-Rotelli committed:
solo las dependencias de produccion en dockere

Javier-Rotelli committed:
WIP

Javier-Rotelli committed:
wip 2

Javier-Rotelli committed:
users in mongo

Javier-Rotelli committed:
courses in mongo

nbarrera committed:
para que mongo no sea accesible desde afuera

solo hay que quitar el expose de mongo

igual se puede acceder desde el nodejs

nbarrera committed:
quito un warning de mongoose

agrego por completitud config en .env
para mongo host y port
al final no los use
igual van con defualt values

nbarrera committed:
Merge branch 'add-mongo' of https://github.com/Javier-Rotelli/desarrollo-seguro-grupo-2 into add-mongo

nbarrera commented on data/courses.json:
```diff
@@ -16,5 +16,11 @@
 		"name": "Diplomatura Desarrollo Seguro de App - 3ra. edición",
 		"initDate": "2023-03-05T03:00:00.000Z",
 		"enrolled": 0
+	},
```
che, a la final estos `data/*.json` hay que borrarlos?
poruqe el contenido ya esta copiado en el entry point init.d de mongo?

nbarrera commented on docker-entrypoint-initdb.d/mongo-init.js:
```diff
@@ -0,0 +1,69 @@
+      "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6Im5iYXJyZXJhIiwiaWF0IjoxNjY5ODY3NTY2LCJleHAiOjE2Njk4Njc4NjZ9.mP6cRYUoyY7XQvFrPSpwylKQV8ttzNAYdonV5wZRPq4BSm9b9QRKiuYDBxlEiDwuFRcRFTPu-JOVO5ZmCXWnnanS3UydXceKmCdT7CHJHlIg7ayWrGrlLa58P8M7u9bFk2NVVeYWN890PDKu3-bVfuOXjPaQak-KGtfORrr_6JRXGNGST9EopCrAfpVQYN7orApIJ10yAIK_QgJGp6s0Q0ez4W_F9Iee--nueKxx25V9vKEv4OOFc9s9wwiSYXesTUckMB5xUZvjJwf1a0MB_M ...
+    username: "rfort",
+    iat: 1669867566,
+    exp: 1669867866,
```
en teoria este token ya expiro el 1 de diciembre...

tendria que probar a ver que onda si lo sigue aceptando..
si lo acepta tengo que arreglarlo y si no lo acepta tendriamos que boletear los tokens  expirados del codigo...

el tema es que el chabon no va a tener ningun token para jugar poruqe no sabemos cuando lo va a corregir.. y la expiration esta bastante corta

sino le mandamos la base sin tokens

nbarrera commented on docker-entrypoint-initdb.d/mongo-init.js:
```diff
@@ -0,0 +1,69 @@
+    _id: "1611d4cb6d0647c2a651b2b2",
+    username: "rfort",
+    password: "$2b$10$2jbjk2fZm9dIwkSfpTDKZuWYcOj4pMzEemIMwdtUNVq3IqvgJgVAa",
+    name: "Ricardo",
```
nunca se sabe
```suggestion
    name: "Raul",
```

nbarrera commented on users/db.js:
```diff
@@ -1,22 +1,30 @@
-  users.push(newUser)
-  writeFileSync(USERS_FILE, JSON.stringify(users.concat(newUser), null, "\t"))
-}
+//export const createUser = ({ password, ...rest } = {}) => {
+//  const newUser = { ...rest, password: hash(password) };
+//  users.push(newUser);
+//  writeFileSync(USERS_FILE, JSON.stringify(users.concat(newUser), null, "\t"));
+//};
```
no dead code
```suggestion
```

nbarrera reviewed (commented):
todo bien,

tendria que probar el tema de los tokens para ver si los tokens vencidos los rechaza
si es asi, me parece que mandaria la base sin tokens, que te parece?

✅  bueno, apruebo pero me falta testear... y me falta solucoinar los conflictos de merge..

asique no merge todavia

nbarrera committed:
documentar como ver la base

nbarrera committed:
requiremento docker

nbarrera committed:
mejoro debug de conexion de db

nbarrera committed:
agrego chequeo de expiration al validar el token

nbarrera committed:
Merge branch 'main' into add-mongo

nbarrera committed:
agrego TODOs

nbarrera committed:
depend on mongo for docker compose

nbarrera committed:
agregando wait-for al docker compose

arreglando que mongo solo exponga el puerto en la red interna de docker-compose

nbarrera committed:
sale ricardo entra raul

nbarrera committed:
no dead code

nbarrera committed:
borrar json data

nbarrera committed:
prettier

nbarrera committed:
fix linter

nbarrera reviewed (approved):
sale con fritas
